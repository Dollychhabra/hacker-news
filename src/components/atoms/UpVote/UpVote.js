import React from 'react';
import PropTypes from 'prop-types';

import Anchor from '../Anchor';
import './UpVote.scss';

const UpVote = ({ vote, objectId }) => {
  return (
    <Anchor
      onClick={e => {
        e.preventDefault();
        vote({
          objectId,
          voted: true,
        });
      }}
      className="up-vote-link"
    >
      <div className="up-vote-arrow" />
    </Anchor>
  );
};

UpVote.propTypes = {
  objectId: PropTypes.string.isRequired,
  vote: PropTypes.func.isRequired,
};

export default UpVote;
