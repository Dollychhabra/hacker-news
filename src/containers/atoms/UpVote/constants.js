export const VOTE_KEY = 'HACKER_NEWS_APP/LIST/VOTE_KEY';
export const VOTE = 'HACKER_NEWS_APP/LIST/VOTE';
