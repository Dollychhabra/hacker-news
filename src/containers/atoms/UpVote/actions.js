import { VOTE } from './constants';

export const vote = data => ({
  type: VOTE,
  data,
});
