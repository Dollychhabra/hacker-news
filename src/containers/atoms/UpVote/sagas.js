/* eslint-disable require-yield */
import { takeLatest } from 'redux-saga/effects';

import { VOTE, VOTE_KEY } from './constants';
import { LocalStorageUtil } from '../../../utils/localStorage';

export function* saveUpVoteSaga(action) {
  const {
    data: { objectId, voted },
  } = action;
  const storage = new LocalStorageUtil();

  if (voted) {
    storage.saveItemInArray(VOTE_KEY, objectId);
  } else {
    storage.removeItemFromArray(VOTE_KEY, objectId);
  }
}

export default function* upVoteSaga() {
  yield takeLatest(VOTE, saveUpVoteSaga);
}
