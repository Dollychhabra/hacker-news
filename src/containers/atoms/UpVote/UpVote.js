import { connect } from 'react-redux';

import { vote } from './actions';

import UpVote from '../../../components/atoms/UpVote/UpVote';

export const mapDispatchToProps = dispatch => {
  return {
    vote: data => {
      dispatch(vote(data));
    },
  };
};

export default connect(
  null,
  mapDispatchToProps,
)(UpVote);
