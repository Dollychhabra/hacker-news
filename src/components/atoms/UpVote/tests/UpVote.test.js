import React from 'react';
import { shallow } from 'enzyme';
import UpVote from '../UpVote';

describe('UnVote component', () => {
  const props = {
    vote: () => {},
    objectId: '233456',
  };

  it('should render correctly and match snapshot', () => {
    const component = shallow(<UpVote {...props} />);
    expect(component).toMatchSnapshot();
  });
});
