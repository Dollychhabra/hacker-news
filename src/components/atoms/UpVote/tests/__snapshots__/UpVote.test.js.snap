// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnVote component should render correctly and match snapshot 1`] = `
<Anchor
  className="up-vote-link"
  onClick={[Function]}
  role="link"
  type=""
>
  <div
    className="up-vote-arrow"
  />
</Anchor>
`;
